src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -186,13 +192,41 @@ public async Task WatchAsync(DotNetWatchContext context, CancellationToken cance
             }
         }
 
-        private static void ConfigureExecutable(DotNetWatchContext context, ProcessSpec processSpec)
+        private void ConfigureExecutable(DotNetWatchContext context, ProcessSpec processSpec)
         {
             var project = context.FileSet.Project;
-            processSpec.Executable = project.RunCommand;
-            if (!string.IsNullOrEmpty(project.RunArguments))
+
+            // For webapps, we can support changes to Program.Main and Startup by restarting the host (aka hot restart).
+            // First determine if we can perform hot restarts.
+            if (!SuppressHotRestart &&
+                ProjectGraph.EntryPointNodes.FirstOrDefault().ProjectInstance is { } projectInstance &&
+                !string.IsNullOrEmpty(projectInstance.GetPropertyValue("UsingMicrosoftNETSdkWeb")))
             {
-                processSpec.EscapedArguments = project.RunArguments;
+                // When performing a hot restart, we have to run a different assembly than the project's output. This
```

Comment:
Can you clarify what this means? Who produces the assembly that is at `TargetPath`?

Reply:
What we want to do is run

```
dotnet exec HotRestart.dll --runtime-config $path/myapp.runtimeconfig.json --deps-json $path/myapp.deps.json
```

TargetPath is the output of building the current project.